examples/6_totp.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+
+###################################
+# WHAT IS IN THIS EXAMPLE?
+#
+# This is a simple TOTP bot that makes use of
+# the team encrypted key-value store, which
+# we interact with using KVStoreClient.
+###################################
+
+import asyncio
+import logging
+import os
+import sys
+from enum import Enum
+
+import pyotp
+
+from pykeybasebot import Bot, EventType
+
+logging.basicConfig(level=logging.DEBUG)
+
+if "win32" in sys.platform:
+    # Windows specific event-loop policy
+    asyncio.set_event_loop_policy(asyncio.WindowsProactorEventLoopPolicy())
+
+
+class TotpMsg(Enum):
+    PROVISION = "provision"
+    REPROVISION = "reprovision"
+    REMOVE = "remove"
+    NOW = "now"
+    URI = "uri"
+    LIST = "list"
+
+
+class TotpHandler:
+    """
+    TotpHandler handles commands sent via chat and uses the team key-value store to
+    provide TOTP client functionality.
+
+    Background: The team encrypted key-value store stores (key, value)
+    pairs, with unique (team, namespace, key)s. The value is encrypted, and the
+    team and namespace are not.
+
+    TotpHandler listens to chat messages of the form:
+
+    `totp {provision|reprovision|remove|now|uri} <issuer>` and
+    `totp list`
+
+    For each provisioned key, the handler stores ("secret", <16 char base32 secret>)
+    in the team's KV store under the namespace of <issuer>. This bot assumes
+    your team's KV store is solely used for TOTP credentials, though this can
+    easily be modified.
+
+    This Keybase bot can be used in place of MFA apps like Google Authenticator and Authy
+    for login processes that require two-step verification. This bot uses the
+    pyotp library, so it is also compatible with other MFA apps.
+
+    A Keybase TOTP bot could be useful for logins where you're already using a strong
+    password over an secure transport, but are required to use TOTP. Or it could be
+    perfect for adding replay attack resistance to the physical keypad lock protecting
+    your team's ice cream freezer.... Note that depending on the login process and
+    how you use this bot with your Keybase team setup, this bot may not be appropriate
+    for threat models that require a physical "second factor" device.
+
+    For more information on TOTP, see https://pyotp.readthedocs.io/en/latest/,
+    https://tools.ietf.org/html/rfc6238.
+
+    This example does minimal error handling.
+    """
+
+    MSG_PREFIX = "totp"
+
+    async def __provision(self, bot, team, issuer, force: bool = False):
+        secret = pyotp.random_base32()
+        if not force:
+            try:
+                # throws exception if 1 is not the latest revision + 1
```

Comment:
my instinct is that this is a lot of complexity for this example. i think it's probably more important, given what this is and where it is, that this code be (1) very easy to understand, (2) a good first pass example of the kvstore than (A) show off the full feature set of kvstore, and (B) be a fully featured TOTP bot. 

not saying we should get rid of any specific functionality here. but if you see something you can cut that's a little complicated and not pretty necessary to understanding basic kvstore functionality, it might be worth a trim. supporting `revision` as an optional parameter kinda feels like that to me.